### PR TITLE
response: fix RES_LINE body data handling

### PR DESCRIPTION
**Pull request by cccs-sadugas**
This PR addresses the bug report #361 and also adds one test case that would fail on `0.5.x`.

The `out_buf` that is consolidated and used to read response data as provided by the user should be cleared to avoid growing indefinitely. This happens when the `RES_LINE` state cannot find a valid response line and treats user provided response data as a response body.

### Testing/QA

I ran `make check` to make sure existing tests still pass.

### CODE REVIEW

File: test/test_main.cpp [hunk 1]
```diff
@@ -2088,3 +2089,60 @@ TEST_F(ConnectionParsing, Expect100) {
...
+    ASSERT_EQ(HTP_RESPONSE_COMPLETE, tx->response_progress);
+
+    struct ResponseBodyDataCallback *user_data = (struct ResponseBodyDataCallback *) htp_tx_get_user_data(tx);
+    ASSERT_TRUE(user_data);
+
+    char *response_body = strndup((const char *) user_data->data, user_data->len);
+    ASSERT_TRUE(response_body);
+
+    EXPECT_EQ(6, user_data->len);
+    EXPECT_STREQ("1\n23\n4", response_body);
```
Comment (catenacyber) [inline test/test_main.cpp:2143]: Should we somehow test that we first get "1 2" then "3 4" ?

File: htp/htp_response.c [hunk 2]
```diff
@@ -1073,6 +1073,7 @@ htp_status_t htp_connp_RES_LINE(htp_connp_t *connp) {
 
                 connp->out_current_consume_offset = connp->out_current_read_offset;
                 htp_status_t rc = htp_tx_res_process_body_data_ex(connp->out_tx, data, len + chomp_result);
+                htp_connp_res_clear_buffer(connp);
```
Comment (catenacyber) [inline htp/htp_response.c:1076]: Nice, did you look for other missing calls to `htp_connp_res_clear_buffer` ?

Comment (cccs-sadugas) [reply htp/htp_response.c:1076]: The `_FINALIZE()` state, which also calls the response body data hook clears this buffer. I haven't looked everywhere though. This should fix the bug we are experiencing.

File: test/test_main.cpp (hunk 1 above)
Comment (catenacyber) [reply test/test_main.cpp:2143]: Should we wait for a better test ?
 
(The fix looks good to me)

Comment (cccs-sadugas) [reply test/test_main.cpp:2143]: You can go ahead and merge this PR while I work on a better test.

**Pull request closed without merging**